app/controllers/users/contacts_controller.rb

```diff
@@ -0,0 +1,40 @@
+module Users
+  class ContactsController < ApplicationController
+    load_resource :user
+    load_and_authorize_resource through: :user
+
+    def index
+      @schools = if @user.group_admin?
+                   @user.school_group.schools.visible.by_name
+                 elsif @user.has_other_schools?
+                   @user.cluster_schools.by_name
+                 else
+                   [@user.school]
+                 end
+      @show_clusters = @schools.any? { |s| s.school_group_cluster.present? }
+      render :index, layout: 'dashboards'
+    end
+
+    # Sometimes creating the contact is failing as its invalid...?
```

Comment:
Do we know why creating is failing? Seems like something that needs ironing out? Is that why you are using Contact.create here and then saving a second time?

Reply:
Will revisit this later, think its related to the active record callbacks